kotlinx-coroutines-core/common/src/CompletableDeferred.kt

```diff
@@ -45,6 +46,17 @@ public interface CompletableDeferred<T> : Deferred<T> {
     public fun completeExceptionally(exception: Throwable): Boolean
 }
 
+/**
+ * Completes this deferred value with the value or exception in the given [result]. Returns `true` if this deferred
+ * was completed as a result of this invocation and `false` otherwise (if it was already completed).
+ *
+ * Subsequent invocations of this function have no effect and always produce `false`.
+ *
+ * This function transitions this deferred in the same ways described by [CompletableDeferred.complete] and
+ * [CompletableDeferred.completeExceptionally].
+ */
+public fun <T> CompletableDeferred<T>.completeWith(result: Result<T>) = result.fold({ complete(it) }, { completeExceptionally(it) })
```

Comment:
Style nit: `fold` better works for functions that return something depending on success/failure, and not so good for side-effecting code like. Stylistically, `onSuccess { complete(it) }.onFailure { completeExceptionaly(it) }` would look better here.

Reply:
But I am returning the same true/false value that complete/completeExceptionally returns.

Reply:
I find the following to be the most readable:

```kotlin
= result.fold(
  onSuccess = { complete(it) }, 
  onFailure = { completeExceptionally(it) }
)
```